### ISSUE TITLE
Docker Desktop program shuts down after a few minutes. Traced it to the Resource Saver as cause.

### ISSUE THREAD

**Issue by davethekay** [docker/for-win#13789]
### Description

On 11/09/2023 I upgraded Windows 10 Docker Desktop to version 4.25.0 (126437). Before upgrade, I had the Resource Saver running and had no issues for months. However, once I upgraded I found that Docker Desktop suddenly terminated after a few minutes.

After doing some recommended fixes such as Clean/Purge data, the Desktop still stopped unexpectedly. I then turned off the Resource Saver option and now it is OK. Had Desktop running now for a whole day with no issues. Has anyone else had this issue?

### Reproduce

1. Start up Docker Desktop on Windows
2. Check the Resource Saver is turned on. Usually this is on by default I believe. If not turn it on
3. Wait a few minutes. After a period of time (a few minutes) Desktop stops running completely

### Expected behavior

Docker Desktop still should be running even when in Resource Saver mode

### docker version

```bash
Docker Desktop: 4.25.0 (126437)
```

### docker info

```bash
Client:
 Version:    24.0.6
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.2-desktop.5
    Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
  compose: Docker Compose (Docker Inc.)
    Version:  v2.23.0-desktop.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.20
    Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.9
    Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scan.exe
  scout: Docker Scout (Docker Inc.)
    Version:  v1.0.9
    Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe
```

### Diagnostics ID

37C39F8A-6893-4836-9659-347B9B533AFD/20231110152345

**Comment by valiko-ua** [comment]
Windows 11, Docker Desktop 4.25.1, the same problem: after some idle time my laptop is automatically locked. After unlocking it, I see docker icon in system tray with resource saver overlay and docker becomes unavailable in WSL.

**Comment by hhromic** [comment]
Same here, two different Windows 10 machines, Docker Desktop 4.25.0 and also 4.25.1 (latest). This did not happen in 4.24.2.
After Docker Desktop enters into Resource Saver mode, the `docker` CLI tool becomes unavailable inside of WSL (Debian).

**Comment by bedla** [comment]
seems as duplicate of https://github.com/docker/for-win/issues/13088 ?

**Comment by bmeier-pros** [comment]
This is not quite the same issue as #13088 - this involves the entire docker desktop application crashing.  The workaround of disabling resource saver fixes this issue as well, but the two issues are different.

**Comment by hhromic** [comment]
I just got notified that a new version (v4.25.2) of Docker Desktop got released.
The [changelog](https://docs.docker.com/desktop/release-notes/#4252) explicitly mentions that this issue was fixed.

I'm running v4.25.2 now and can confirm it was fixed for me 👍.

**Comment by oleksis** [comment]
This issue was the inspiration for [Moboker](https://github.com/oleksis/Mobocker#images-powered-by-dalle-3): "Happy Mobockerc, your services never sleep!"

**Comment by bangbaew** [comment]
I use Hyper-V and this happens to me as well, if I enable Resource Saver, the Docker Engine goes down after idle and couldn't be turned on again until I restart my computer.

**Comment by MihaelaStoica** [comment]
We've released a new version of Docker Desktop, 4.25.2, containing a fix for this issue, see [the release notes](https://docs.docker.com/desktop/release-notes/).
Closing this issue for now, but if the issue persists, please let us know.

**Comment by subhanlicygr-arch** [comment]
link
